pkg/build/controller/strategy/custom.go

```diff
@@ -57,7 +57,10 @@ func (bs *CustomBuildStrategy) CreateBuildPod(build *buildv1.Build, additionalCA
 		return nil, fmt.Errorf("failed to encode the build: %v", err)
 	}
 
-	containerEnv := []corev1.EnvVar{{Name: "BUILD", Value: string(data)}}
+	containerEnv := []corev1.EnvVar{
+		{Name: "BUILD", Value: string(data)},
+		{Name: "LANG", Value: "en_US.utf8"},
```

Comment:
Set the `LANG` environment variable for all containers in the build pod.

Binary builds use `bsdtar` to extract the content from STDIN. `LANG` needs to be set in order for the bsdtar to extract UTF-8 filenames.